api/util/ipnet/ipnet.go

```diff
@@ -16,6 +16,8 @@ var nullBytes = []byte(nullString)
 // IPNet wraps net.IPNet to get CIDR serialization.
 //
 // +kubebuilder:validation:Type=string
+// +kubebuilder:validation:MaxLength=43
+// +kubebuilder:validation:XValidation:rule=`self.matches('^((\\d{1,3}\\.){3}\\d{1,3}/\\d{1,2})$') || self.matches('^([0-9a-fA-F]{0,4}:){2,7}([0-9a-fA-F]{0,4})?/[0-9]{1,3}$')`,message="cidr must be a valid IPv4 or IPv6 CIDR notation (e.g., 192.168.1.0/24 or 2001:db8::/64)"
```

Comment:
CEL has a CIDR parsing library, it's in OpenShift API servers from 4.16 onwards, you could use that instead of the pattern

Reply:
Discussed offline this is available in vanilla kube from 1.31 API servers and in Openshift from 4.16.
There are use cases that consume hypershift using non Openshift versions before 1.31.
So I'll punt on this for now and add a todo
@JoelSpeed Do you have a link to the proposal where support for this was added in upstream?

Reply:
No design doc no, but added in https://github.com/kubernetes/kubernetes/pull/121912